### ISSUE TITLE
Module does not handle colons ':' in values well

### ISSUE THREAD

**Issue by i7c** [keirans/azuremetadata#7]
Hi,

thanks for this module. Though, I’m having an issue if I put a tag like
host : localhost:1234

I *think* the clue is the : in the value of the tag. `puppet facts` reports 
"Error: Facter: error while resolving custom fact "az_metadata": invalid number of elements (3 for 1..2)"

and the az_metadata section is missing in the facts.

**Comment by keirans** [comment]
Hi There,
Thanks for raising this, I'll take a look, however I can't give you an ETA on this at the moment because of other priorities.

I'll hopefully get some time to look at it in the next week or so.

Cheers,

K

**Comment by keirans** [comment]
Confirmed as an issue and resolved in version 0.1.4.

This will be available on the forge shortly.

Thanks for raising this, and apologies for the delay in getting this fixed.

K

```
# facter az_metadata
{
  compute => {
    location => "australiasoutheast",
    name => "Keiran-PuppetDev",
    offer => "RHEL",
    osType => "Linux",
    placementGroupId => "",
    platformFaultDomain => "0",
    platformUpdateDomain => "0",
    publisher => "RedHat",
    resourceGroupName => "keiran-puppetdev",
    sku => "7-RAW",
    subscriptionId => "0432b1d0-5e2e-4e2a-ad73-e33d0652e3f7",
    tags => {
      Exampletag => "examplevalue",
      badtag => "part1:part2"
    },
    version => "7.4.2018010506",
    vmId => "20ca74fd-5c9f-442f-afe4-5140b357de6a",
    vmSize => "Standard_D2s_v3"
  },
  network => {
    interface => [
      {
        ipv4 => {
          ipAddress => [
            {
              privateIpAddress => "10.0.2.4",
              publicIpAddress => "13.70.186.81"
            }
          ],
          subnet => [
            {
              address => "10.0.2.0",
              prefix => "24"
            }
          ]
        },
        ipv6 => {
          ipAddress => []
        },
        macAddress => "000D3AE139A0"
      }
    ]
  }
}
```

**Comment by keirans** [comment]
I've pushed this to forge and validated all is functional.

```
# puppet module install keirans-azuremetadata
Notice: Preparing to install into /etc/puppetlabs/code/environments/production/modules ...
Notice: Downloading from https://forgeapi.puppet.com ...
Notice: Installing -- do not interrupt ...
/etc/puppetlabs/code/environments/production/modules
└── keirans-azuremetadata (v0.1.4)
#

# cat  /tmp/testing.pp
$facts['az_metadata']['compute']['tags'].each | $value | {
  notice("Tag: ${value[0]} Value: ${value[1]}")
}
# puppet apply /tmp/testing.pp
Notice: Scope(Class[main]): Tag: Exampletag Value: examplevalue
Notice: Scope(Class[main]): Tag: badtag Value: part1:part2
Notice: Compiled catalog for keiran-puppetdev.10rhifqqxd2u3mquj1w0huiara.qx.internal.cloudapp.net in environment production in 0.03 seconds
Notice: Applied catalog in 0.08 seconds
#
```
